src/main/scala/is/hail/io/vcf/HtsjdkRecordReader.scala

```diff
@@ -85,152 +85,6 @@ abstract class HtsjdkRecordReader[T] extends Serializable {
   def genericGenotypes: Boolean
 }
 
-case class GenotypeRecordReader(vcfSettings: VCFSettings) extends HtsjdkRecordReader[Genotype] {
```

Comment:
Are you intentionally keeping the one subclass `GenericRecordReader` separate rather than adding the one method `readRecord` back to `HtsjdkRecordReader`?